### DISCUSSION TITLE
Conhecendo compiladores: analisador léxico

### DISCUSSION THREAD

**Discussion by mlarasusan** [gittogethers/octogatosconf22#114] [Talks in Portuguese]
**Palestrante**: Bruna Ferreira
**Assunto**: Um compilador possui vários processos, um dos processos iniciais é a análise léxica do código escrito pela pessoa programadora. Todo código é analizado, caractere por caractere, retornando algo válido ou não, de acordo com as regras da linguagem. Vamos aprender como ele funciona?

**Comment by lissarubi** [comment]
Um conjunto de números e letras são uma sequência de letras? tipo `nome01tabela04`

**Reply by bugelseif** [reply]
Nesse caso seria alfanumérico, no contexto seria aceito pela regra:
D = [0-9]
L = [a-zA-Z]
ID = L(L|D)*  --> letra + (sequencia de 0 ou mais letras OU números)
ID = n + (ome01tabela04)

**Reply by lissarubi** [reply]
ahhhhhh, obrigada bu :heart:

**Comment by bugelseif** [comment]
Compilador:
https://github.com/bugelseif/compilador

Twitter:
https://twitter.com/bug_elseif
